### PR TITLE
fix: register class should not only depend on name

### PR DESCRIPTION
**Pull request by JoanFM**
Linked to #1670  and #1527

### CODE REVIEW

**Comment by tadejsv** [comment]
One question here, not really related to the PR but more of a fundamental python question: how is it possible that `jina.hub.encoders.nlp.TransformerTorchEncoder` (and litreally every other encoder) was registered? I checked and it is never referenced/imported anywhere (the tests will only import the local one).

**Comment by JoanFM** [comment]
> One question here, not really related to the PR but more of a fundamental python question: how is it possible that `jina.hub.encoders.nlp.TransformerTorchEncoder` (and litreally every other encoder) was registered? I checked and it is never referenced/imported anywhere (the tests will only import the local one).

I believe is some prework done as u can see when u do jina check.

**Comment by tadejsv** [comment]
> > One question here, not really related to the PR but more of a fundamental python question: how is it possible that `jina.hub.encoders.nlp.TransformerTorchEncoder` (and litreally every other encoder) was registered? I checked and it is never referenced/imported anywhere (the tests will only import the local one).
> 
> I believe is some prework done as u can see when u do jina check.

You're right, I found that `__init__.py` file executes this: 
``` python
_ji.import_classes('jina.hub', show_import_table=False, import_once=True)
```

**Comment by tadejsv** [comment]
@JoanFM I think this problem (of registring a class with only the name) also appears in some other places - definitely for drivers. I found those instances by searching for `__name__`. Probably would be best to solve it there as well?

**Comment by JoanFM** [comment]
> @JoanFM I think this problem (of registring a class with only the name) also appears in some other places - definitely for drivers. I found those instances by searching for `__name__`. Probably would be best to solve it there as well?

these MetaClass is only used for Executors, so at least for storing init kwargs should be okey.

File: tests/unit/test_exectype.py [hunk 1]
```diff
@@ -28,18 +40,21 @@ def assert_bi():
     # we override BaseIndexer now, without force it shall not store all init values
```
Comment (nan-wang) [inline tests/unit/test_exectype.py:40]: Would you please change the comments here as well? By this point, `BaseIndexer` has not registered in `reg_cls_set` yet and `store_init_kwargs` will be executed

**Review by nan-wang — CHANGES_REQUESTED** [review]
LGTM👍

File: tests/unit/test_exectype.py (hunk 1 above)
Comment (JoanFM) [reply tests/unit/test_exectype.py]: Done!

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**